tests/test_buffer_dynamic.py

```diff
@@ -931,6 +932,45 @@ def test_bufferPoolPercentage(self, dvs, testlog):
 
             real_percentage = percentage_size * 100 / original_size
             assert abs(percentage - real_percentage) < margin
+
+            # Test 2: Test the buffer pool calculation with a port with multiple queues
+            # Store existing Ethernet0 entries for restoration
+            original_eth0_entries = {}
+            eth0_keys = self.config_db.get_keys('BUFFER_QUEUE')
+            for key in eth0_keys:
+                if key.startswith('Ethernet0|'):
+                    original_eth0_entries[key] = self.config_db.get_entry('BUFFER_QUEUE', key)
+                    self.config_db.delete_entry('BUFFER_QUEUE', key)
+
+            # Startup port
+            dvs.port_admin_set('Ethernet0', 'up')
+
+            # Create buffer profile
+            self.config_db.update_entry('BUFFER_PROFILE', 'egress_test_profile',
+                                      {'dynamic_th': '0',
+                                       'pool': 'egress_lossy_pool',
+                                       'size': '16384'})
+
+            # Create buffer queue entries
+            self.config_db.update_entry('BUFFER_QUEUE', 'Ethernet0|0-7', {'profile': 'egress_test_profile'})
+            self.config_db.update_entry('BUFFER_QUEUE', 'Ethernet0|8-12', {'profile': 'egress_test_profile'})
+            self.config_db.update_entry('BUFFER_QUEUE', 'Ethernet0|13-19', {'profile': 'egress_test_profile'})
+
+            # Run lua script and check output
+            _, output = dvs.runcmd("redis-cli --eval /usr/share/swss/buffer_pool_vs.lua")
+            assert re.search(r"debug:BUFFER_PROFILE_TABLE:egress_test_profile:16384:20", output), "Profile reference count not found in output"
```

Comment:
Can we also double check ASIC_DB?

Reply:
Hi @kperumalbfn 
Understanding the mapping between ASIC_DB and APPL DB requires more complicated logic which we do not have in vs.
We have a lot of checks between ASIC/APPL DB in the sonic-mgmt test leveraging existing infra there
So, I don't want to add such logic here.
What do you think?

Reply:
Can we use counter_db's QUEUE_NAME_MAP to get the queue OID based on the port+queueIndex and validate with ASIC_DB? This will make sure that profiles are attached to the queues 10-19

Reply:
Hi @kperumalbfn 
We do not need to do it because we didn't change the queue mappings or whether the buffer profile is applied to queue 10-19. That is guaranteed by buffer orch.
We just corrected a scenario where there are two digits in BUFFER_QUEUE table and translated it to a correct number.
eg, if we have Ethernet8|7-15, it should be 9 queues. But it got the wrong result without this change.